topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -473,6 +478,43 @@ public void setTopicCount(int topics) {
                 action.execute().onComplete(actionResult -> {
                     LOGGER.debug("{}: Executing handler for action {} on topic {}", logContext, action, lockName);
                     action.result = actionResult;
+
+                    String keytag = namespace + ":" + "KafkaTopic" + "/" + key.asKubeName().toString();
+
+                    Optional<Meter> metric = metrics.meterRegistry().getMeters()
+                            .stream()
+                            .filter(meter -> meter.getId().getName().equals(METRICS_PREFIX + "resource.state") &&
+                                    meter.getId().getTags().contains(Tag.of("kind", "KafkaTopic")) &&
+                                    meter.getId().getTags().contains(Tag.of("name", key.asKubeName().toString())) &&
+                                    meter.getId().getTags().contains(Tag.of("resource-namespace", namespace))
+                            ).findFirst();
+
+                    if (metric.isPresent()) {
+                        // remove metric so it can be re-added with new tags
+                        metrics.meterRegistry().remove(metric.get().getId());
+                        resourcesStateCounter.remove(keytag);
+                        LOGGER.debug("{}: Removed metric " + METRICS_PREFIX + "resource.state{}", action.topic, keytag);
+                    }
+
+                    if (action.topic != null) {
+
+                        boolean succeeded = actionResult.succeeded();
+
+                        Tags metricTags;
+                        metricTags = Tags.of(
+                                Tag.of("kind", action.topic.getKind()),
+                                Tag.of("name", action.topic.getMetadata().getName()),
+                                Tag.of("resource-namespace", namespace),
+                                Tag.of("reason", succeeded ? "none" : actionResult.cause().getMessage() == null ? "unknown error" : actionResult.cause().getMessage()));
+
+                        resourcesStateCounter.computeIfAbsent(keytag, tags ->
+                                metrics.gauge(METRICS_PREFIX + "resource.state", "Current state of the resource: 1 ready, 0 fail", metricTags));
+
+                        resourcesStateCounter.get(keytag).set(succeeded ? 1 : 0);
+
+
+                        LOGGER.debug("{}: Updated metric " + METRICS_PREFIX + "resource.state{} = {}", action.topic, metricTags, succeeded ? 1 : 0);
+                    }
```

Comment:
`metrics` already maintains a mapping of `name+tags` to `metric`, so there should be no need to maintain `resourcesStateCounter` yourself. It turns out that re-registering an existing metric (which is what `metrics.gauge` does under the covers) will return the existing metric if it was already registered, so I _think_ all we need to do here is create-or-get the gauge and then set its value:

```suggestion
                        Tags metricTags = Tags.of(
                                Tag.of("kind", action.topic.getKind()),
                                Tag.of("name", action.topic.getMetadata().getName()),
                                Tag.of("resource-namespace", namespace),
                                Tag.of("reason", succeeded ? "none" : actionResult.cause().getMessage() == null ? "unknown error" : actionResult.cause().getMessage()));

                        metrics.gauge(METRICS_PREFIX + "resource.state", "Current state of the resource: 1 ready, 0 fail", metricTags).set(actionResult.succeeded() ? 1 : 0);
```

Reply:
That said, I would still factor the above code into an `updateReconciliationGauge()` method

Reply:
Also give yourself a `TOPIC_STATE_GAUGE` constant to save yourself having to write `METRICS_PREFIX + "resource.state"` everywhere.

Reply:
@tombentley Hi. I have applied the susgestion. I tried using this name `TOPIC_STATE_GAUGE` but it was giving me some checkstyle issues.

Reply:
What kind of checkstyle issue did you get?

Reply:
``` Name 'TOPIC_STATE_GAUGE' must match pattern '^[a-z][a-zA-Z0-9]*$'. [MemberName]``` This is the error I am getting.

Reply:
I searched about it, It says the string must be in camel case